pointcloud/basic_octree_utils_test.go

```diff
@@ -327,3 +330,63 @@ func printBasicOctree(logger golog.Logger, bOct *BasicOctree, s string) {
 		printBasicOctree(logger, v, s+"-+-")
 	}
 }
+
+// Test the functionalities involved with converting a pointcloud into a basic octree.
+func TestBasicOctreeCollision(t *testing.T) {
+	startPC, err := makeFullPointCloudFromArtifact(
+		t,
+		"slam/example_cartographer_outputs/viam-office-02-22-1/pointcloud/pointcloud_0.pcd",
+		BasicType,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	center := getCenterFromPcMetaData(startPC.MetaData())
+	maxSideLength := getMaxSideLengthFromPcMetaData(startPC.MetaData())
+
+	basicOct, err := NewBasicOctree(center, maxSideLength)
+	test.That(t, err, test.ShouldBeNil)
+
+	startPC.Iterate(0, 0, func(p r3.Vector, d Data) bool {
+		_, _, blueProb := d.RGB255()
+		d.SetValue(int(blueProb))
+		if err = basicOct.Set(p, d); err != nil {
+			return false
+		}
+		return true
+	})
+
+	test.That(t, startPC.Size(), test.ShouldEqual, basicOct.Size())
+
+	// create a non-colliding obstacle far away from any octree point
+	far, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{1, 2, 3}, "far")
+	test.That(t, err, test.ShouldBeNil)
+	// create a non-colliding obstacle near an octree point
+	near, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{-2443, 0, 3855}), r3.Vector{1, 2, 3}, "near")
+	test.That(t, err, test.ShouldBeNil)
+	// create a colliding obstacle overlapping an octree point
+	hit, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{-2443, 0, 3855}), r3.Vector{12, 2, 30}, "hit")
```

Comment:
[minor] Breaking these up into t.Runs() would be helpful for analyzing failures

Reply:
Done